consensus/bor/bor.go

```diff
@@ -399,6 +399,22 @@ func (c *Bor) verifyCascadingFields(chain consensus.ChainHeaderReader, header *t
 		return err
 	}
 
+	if (number+1)%c.config.Sprint == 0 {
+		newValidators, err := c.GetCurrentValidators(header.ParentHash, number+1)
```

Comment:
I am not sure if this works in full sync mode. That was the reason we tweaked it to check validators from previous header.

Reply:
I think you mean the check of `validateHeaderExtraField`, but it only checks length, doesn't validate against the actual validators.